File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        n=len(nums)
        l=0
        r=n-1
        max_water=0
        while l<=r:
            max_water=max(max_water,(min(nums[l],nums[r])*(r-l)))
            if(nums[l]<=nums[r]):
                l+=1
            else:
                r-=1
        return max_water       
        

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int n=nums.size();
        int l=0,r=n-1;
        int max_water=0;
        while(l<r){
            max_water=max(max_water,(min(nums[l],nums[r])*(r-l)));
            if(nums[l]<=nums[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return max_water;
        
    }
};